### PR TITLE
Deflake the BinaryWebSocketSpec

### LINKED ISSUE #8718: Remove flakiness of BinaryWebSocketSpec
This test `io.micronaut.http.server.netty.websocket.BinaryWebSocketSpec` `test binary websocket exchange` is flaky when executed in a suite :

https://ge.micronaut.io/scans/tests?search.timeZoneId=Europe/Madrid&tests.container=io.micronaut.http.server.netty.websocket.BinaryWebSocketSpec&tests.test=test%20binary%20websocket%20exchange

### PR DESCRIPTION
**Pull request by timyates**
Step 1. Remove the Retry annotation as Flakey test detection checks this and it makes it hard to see the failure. (also, nested retry annotations smell bad)

Trying to see it fail locally with

```
while ./gradlew :http-server-netty:test --tests BinaryWebSocketSpec; do :; done
```

[Got one](https://ge.micronaut.io/s/ah74fuy6zujg2/tests/overview?outcome=failed)

[After a long time](https://ge.micronaut.io/scans?search.rootProjectNames=micronaut&search.startTimeMax=1675699159532&search.startTimeMin=1675641600000&search.tasks=:http-server-netty:test&search.timeZoneId=Europe/London&search.usernames=tim)...

### CODE REVIEW

File: http-server-netty/src/test/groovy/io/micronaut/http/server/netty/websocket/BinaryChatClientWebSocket.java [hunk 1]
```diff
@@ -70,9 +70,8 @@ public WebSocketSession getSession() {
     }
 
     @OnMessage
-    public void onMessage(
-            byte[] message) {
-        System.out.println("Client received message = " + new String(message));
+    public void onMessage(byte[] message) {
+        System.out.println("Client for " + username + " on topic " + topic + " received message = " + new String(message));
```
Comment (sdelamo) [inline http-server-netty/src/test/groovy/io/micronaut/http/server/netty/websocket/BinaryChatClientWebSocket.java]: delete print statement?

**Review by sdelamo — CHANGES_REQUESTED** [review]
Is this ready for review?

**Comment by timyates** [comment]
No, it's not ready for review.

It's still flaky

So far my findings are as follows:

1. When "bob" disconnects, the io.micronaut.http.server.netty.websocket.BinaryChatServerWebSocket#onClose is called
2. The message is sent with no errors
3. It is never received by io.micronaut.http.netty.websocket.AbstractNettyWebSocketHandler#handleWebSocketFrame

It *must* be a race condition somewhere, but I cannot currently see how or where

@graemerocher anything jump to mind?

File: http-server-netty/src/test/groovy/io/micronaut/http/server/netty/websocket/BinaryChatClientWebSocket.java (hunk 1 above)
Comment (timyates) [reply http-server-netty/src/test/groovy/io/micronaut/http/server/netty/websocket/BinaryChatClientWebSocket.java]: These are there to try and see what is going on...  Once we work this out, they can and will be removed

**Comment by graemerocher** [comment]
perhaps the connection is closed before the message can be delivered

**Comment by timyates** [comment]
From my tests, it appears not... 🤔

**Comment by timyates** [comment]
Right, I had that running in a loop for a good 20 minutes with no failures...

It seems to be sending synchronous messages from the ServerWebSocket that sometimes results in them being dropped

**Review by graemerocher — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**